### PR TITLE
88279: Backend Transform Utils Refactor

### PR DESCRIPTION
**Pull request by maxx1128**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Summary

- *This work is behind a feature toggle (flipper): YES*

This is part of several back-end transform refactoring tickets. This one moves over several non-class-specific methods to the utils module.

## Related issue(s)

- *https://github.com/department-of-veterans-affairs/va.gov-team/issues/88279*

## Testing done

Covered by existing tests

## What areas of the site does it impact?

VA Debt backend transform

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [x]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x]  Feature/bug has a monitor built into Datadog (if applicable)
- [x]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected

### CODE REVIEW

**Review by digitaldrk — APPROVED** [review]

File: modules/debts_api/lib/debts_api/v0/fsr_form_transform/utils.rb
Comment (stevenjcumming) [inline modules/debts_api/lib/debts_api/v0/fsr_form_transform/utils.rb:3]: This is a great refactor. I have one suggestion that I think will work but you would want to test to confirm. 

Instead of `re_camel`, `re_camel_array`, and `re_camel_hash` you can combine those into one method and then remove the `rubocop:disable`

```ruby
def re_camel(x)
  case x
  when Array
    x.map { |el| re_camel(el) }
  when Hash
    x.transform_keys { |key| key.camelcase(:lower) }.transform_values { |val| re_camel(val) }
  else
    x
  end
end
```

Comment (maxx1128) [reply modules/debts_api/lib/debts_api/v0/fsr_form_transform/utils.rb:3]: This is a great idea! I'll try it out.

**Comment by maxx1128** [comment]
I'm aiming to get this merged before the end of the sprint today. All the tests are finally passing, so they just need a final review from @stevenjcumming or @digitaldrk before merge ✌️

**Review by digitaldrk — APPROVED** [review]

**Review by stevenjcumming — APPROVED** [review]

**Comment by stevenjcumming** [comment]
@maxx1128 approved

**Pull request merged**
